libyul/AsmParser.cpp

```diff
@@ -507,21 +507,9 @@ TypedName Parser::parseTypedName()
 YulString Parser::expectAsmIdentifier()
 {
 	YulString name{currentLiteral()};
-	switch (currentToken())
-	{
-	case Token::Return:
-	case Token::Byte:
-	case Token::Address:
-	case Token::Bool:
-	case Token::Identifier:
-	case Token::Var:
-	case Token::In:
-		break;
-	default:
+	const auto token = currentToken();
+	if (token != Token::Identifier && TokenTraits::isYulToken(token))
```

Comment:
I've realised there is another switch statement with the same set of exceptions.

Reply:
Yeah, there should be at least two, maybe more.

Reply:
Only found two places in total, but changing the second is a bit weird -- pushed.